queue/abstract/queue_state.lua

```diff
@@ -0,0 +1,127 @@
+-- Queue states.
+
+local log   = require('log')
+local fiber = require('fiber')
+
+--[[ States switching scheme:
+
+                       +-----------+
+                       |  RUNNING  |
+                       +-----------+
+                        ^         | (rw -> ro)
+      (ro -> rw)        |         v
++------+       +---------+       +--------+
+| INIT | --->  | STARTUP |       | ENDING |
++------+       +---------+       +--------+
+                        ^         |
+             (ro -> rw) |         v
+                       +-----------+
+                       |  WAITING  |
+                       +-----------+
+]]--
+
+local queue_state = {
+    states = {
+        INIT    = 'i',
+        STARTUP = 's',
+        RUNNING = 'r',
+        ENDING  = 'e',
+        WAITING = 'w',
+    }
+}
+
+local current = queue_state.states.INIT
+
+local function get_state_key(value)
+    for k, v in pairs(queue_state.states) do
+        if v == value then
+            return k
+        end
+    end
+
+    return nil
+end
+
+local function max_lag()
+    local max_lag = 0
+
+    for i=1, #box.info.replication do
```

Comment:
```lua
tarantool> box.info.replication
---
- 1:
    id: 1
    uuid: 84560a97-1cba-4e24-8814-d49fa1f9d6a5
    lsn: 1001048
  3:
    id: 3
    uuid: 2d82e625-3a6c-427a-9075-93bfc5c07dd7
    lsn: 0
...

tarantool> #box.info.replication
---
- 1
...

tarantool> table.maxn(box.info.replication)
---
- 3
...
```

Use `table.maxn()` for possibly holed arrays.

Reply:
Fixed